pg_shard.c

```diff
@@ -625,8 +664,7 @@ PlanSequentialScan(Query *query, int cursorOptions, ParamListInfo boundParams)
 	enable_indexscan = false;
 	enable_bitmapscan = false;
 
-	queryCopy = copyObject(query);
-	sequentialScanPlan = standard_planner(queryCopy, cursorOptions, boundParams);
+	sequentialScanPlan = standard_planner(query, cursorOptions, boundParams);
```

Comment:
`BuildLocalQuery` creates a copy for us, but I didn't want to put its logic inside of `PlanSequentialScan`. So we're left with either:

  * Having two calls to `copyObject` even though only one is needed
  * Moving `BuildLocalQuery`'s logic inside of `PlanSequentialScan`, which complicates the signature and single responsibility principle.

What if I just change the function comment on `PlanSequentialScan` to document that it expects to "own" `query` and will modify it?